app/controllers/AmendDeclarationController.scala

```diff
@@ -16,40 +16,38 @@
 
 package controllers
 
+import config.featureFlags.DeclarationAmendmentsConfig
+import connectors.CustomsDeclareExportsConnector
 import controllers.actions.{AuthAction, VerifiedEmailAction}
-import forms.Lrn
+import controllers.declaration.routes.SummaryController
+import controllers.routes.RootController
 import handlers.ErrorHandler
-import models.requests.{AuthenticatedRequest, ExportsSessionKeys}
-import play.api.Logging
-import play.api.i18n.I18nSupport
+import models.requests.ExportsSessionKeys
 import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
-import uk.gov.hmrc.play.bootstrap.controller.WithUnsafeDefaultFormBinding
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
 
 import javax.inject.Inject
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
 
 class AmendDeclarationController @Inject() (
   authenticate: AuthAction,
   verifyEmail: VerifiedEmailAction,
   errorHandler: ErrorHandler,
-  mcc: MessagesControllerComponents
-) extends FrontendController(mcc) with I18nSupport with Logging with WithUnsafeDefaultFormBinding {
+  mcc: MessagesControllerComponents,
+  declarationAmendmentsConfig: DeclarationAmendmentsConfig,
+  connector: CustomsDeclareExportsConnector
+)(implicit ec: ExecutionContext) extends FrontendController(mcc) {
 
-  def displayPage: Action[AnyContent] = (authenticate andThen verifyEmail).async { implicit request =>
-    getSessionData() match {
-      case Some((_, mrn, lrn, ducr)) => Future.successful(Ok(""))
-      case _                         => errorHandler.displayErrorPage
+  val submit: Action[AnyContent] = (authenticate andThen verifyEmail).async { implicit request =>
+    if (!declarationAmendmentsConfig.isEnabled) Future.successful(Redirect(RootController.displayPage))
+    else request.session.get(ExportsSessionKeys.submissionId) match {
+      case Some(submissionId) =>
+        connector.findOrCreateDraftForAmend(submissionId).map { declarationId =>
+          Redirect(SummaryController.displayPage)
+            .addingToSession(ExportsSessionKeys.declarationId -> declarationId)
+        }
```

Comment:
We still have a few "submission" keys in Session at this point. Do we need to remove them or we do not care to bring them forward on the next page?

Reply:
If I'm understanding you correctly, keeping these session keys active should be fine as this new draft amend declaration is associated with the current submissionId (and will have the same DUCR, LRN and MRN values as it's parent). 
This however is opposed to when you create a completely new declaration by 'copying' an existing declaration. Then the new declaration will have a different submissionId & MRN (once it is submitted) and certainly will have a different DUCR and LRN. In that scenario these session keys should definitely be cleared. :+1: